### ISSUE TITLE
Bug: Multiple calls to the same thread with different parameters produce the same choice result.

### ISSUE THREAD

**Issue by Udellgames** [inkle/ink#167]
When running the same thread twice with different parameters, choosing an option from the first thread results in the option from the second thread being executed.

**Example file:**
```
VAR player_location = "cavern entrance"
-> area

=area
You are standing at the {player_location}.
{
    -   player_location == "cavern entrance": -> cavern_entrance
    -   player_location == "market": -> market
    -   player_location == "ceiling's edge": -> ceilings_edge
}

=cavern_entrance
<- navigation("market", "north")
<- navigation("ceiling's edge", "south")

=market
<- navigation("market", "north")
<- navigation("ceiling's edge", "south")

=ceilings_edge
<- navigation("market", "north")
<- navigation("ceiling's edge", "south")

=navigation(area_name, direction)
{& In the|To the} {direction} {& lies the {area_name}|is the {area_name}| the {area_name} can be found}.
+ Go {&to the|} {direction}.
    ~player_location = area_name
    -> area
```

**Output:**
```
You are standing at the cavern entrance.

In the north lies the market.

To the south is the ceiling's edge.
```
* Go to the north.
* Head south.

**Player picks go north**

```
Head south.

You are standing at the ceiling's edge.

In the north the market can be found.

To the south lies the ceiling's edge.
```

Inky version used: **0.4 win32-x64**, Ink version: **0.4.1**

**Comment by joethephish** [comment]
Just tried to repro this, and it seems to be working fine in a development build, so the issue must've been resolved as part of another bug fix (at a guess, perhaps this fix to do with temporaries: cff539c451ca7fa0b6061a3ae6bfb2fdab5ddbe7, thanks again @cduquesne!)

Will put out a new release of ink and inky soon with the fix integrated.

**Comment by joethephish** [comment]
And here it is! https://github.com/inkle/inky/releases/latest
